analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -723,29 +728,61 @@ def add_arguments_to_parser(parser):
         func=main, func_process_config_file=cmd_config.process_config_file)
 
 
-def __get_skip_handler(args) -> skiplist_handler.SkipListHandler:
+def get_source_file_paths(
+    file_filters: List[str],
+    compile_commands: tu_collector.CompilationDB,
+    header_file_extensions=(
+        '.h', '.hh', '.H', '.hp', '.hxx', '.hpp', '.HPP', '.h++', '.tcc')
+) -> List[str]:
     """
-    Initialize and return a skiplist handler if
-    there is a skip list file in the arguments or files options is provided.
+    Returns a list of source files for existing header file otherwise returns
+    with the same file path expression.
     """
-    skip_file_content = ""
+    file_paths = []
+    for file_filter in file_filters:
+        file_paths.append(file_filter)
+
+        if os.path.exists(file_filter) and \
+                file_filter.endswith(header_file_extensions):
+            LOG.info("Get dependent source files for '%s'...", file_filter)
+            dependent_sources = tu_collector.get_dependent_sources(
+                compile_commands, file_filter)
+
+            LOG.info("Get dependent source files for '%s' done.", file_filter)
+            LOG.debug("Dependent source files: %s",
+                      ', '.join(dependent_sources))
+
+            file_paths.extend(dependent_sources)
+
+    return file_paths
+
 
+def __get_skip_handlers(args, compile_commands) -> SkipListHandlers:
+    """
+    Initialize and return a list of skiplist handlers if
+    there is a skip list file in the arguments or files options is provided.
+    """
+    skip_handlers = SkipListHandlers()
```

Comment:
Wouldn't concatenation of two skipfile contents be equivalent with the usage of `SkipListHandlers()`? Because I think it wouldn't be necessary introducing `SkipListHandlers()` if they are.

Reply:
I described the problem why we can't use a single skip file in the following issue: https://github.com/Ericsson/codechecker/issues/3595#issuecomment-1046598309